src/main/java/uk/gov/ida/verifyserviceprovider/configuration/HubMetadataConfiguration.java

```diff
@@ -36,26 +34,16 @@
     private final String jerseyClientName;
     private final boolean shouldLoadTrustStoreFromResources;
 
-    @JsonCreator
-    public MetadataConfigurationWithHubDefaults(
-        @JsonProperty("trustStorePath") String trustStorePath,
-        @JsonProperty("trustStorePassword") String trustStorePassword,
-        @JsonProperty("uri") URI uri,
-        @JsonProperty("minRefreshDelay") Long minRefreshDelay,
-        @JsonProperty("maxRefreshDelay") Long maxRefreshDelay,
-        @JsonProperty("expectedEntityId") String expectedEntityId,
-        @JsonProperty("jerseyClientConfiguration") JerseyClientConfiguration jerseyClientConfiguration,
-        @JsonProperty("jerseyClientName") String jerseyClientName
-    ) {
-        this.trustStorePath = ofNullable(trustStorePath).orElseGet(() -> generateTrustStorePath(uri));
-        this.trustStorePassword = ofNullable(trustStorePassword).orElse(DEFAULT_TRUST_STORE_PASSWORD);
-        this.uri = uri;
-        this.minRefreshDelay = ofNullable(minRefreshDelay).orElse(60000L);
-        this.maxRefreshDelay = ofNullable(maxRefreshDelay).orElse(600000L);
-        this.expectedEntityId = generateExpectedEntityId(uri, expectedEntityId);
-        this.jerseyClientConfiguration = ofNullable(jerseyClientConfiguration).orElse(createClient());
-        this.jerseyClientName = ofNullable(jerseyClientName).orElse(HUB_JERSEY_CLIENT_NAME);
-        this.shouldLoadTrustStoreFromResources = (trustStorePath == null);
+    public HubMetadataConfiguration(HubEnvironment environment, MetadataConfigurationOverrides overrides) {
```

Comment:
Do we have enough `@Jsonthings` for it to work out about the environment / overrides things? I thought we'd need the `@JsonProperty / @JsonCreator` annotations. I guess not though, otherwise it wouldn't work 🤔

Reply:
This isn't created directly from JSON anymore, it's called by the VerifyHubConfiguration class after that has deserialised both the environment and the overrides. (That class also needs the environment to do a similar thing for the hubSsoLocation)